### PR TITLE
[Angular] Upgrade to v17

### PR DESCRIPTION
**Pull request by sc-addypathania**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [x] Upgrade guide entry added

## Description / Motivation
- Update Angular and core dependencies to ~17.3.11
- Updated Typescript to ~5.2.2
- Update `zone.js/dist/zone-node` to `zone.js`

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by yavorsk — APPROVED** [review]
There are zone.js imports in templates\angular\src\tests.ts should we update those too?
otherwise looks good to me!

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,10 @@ Our versioning strategy is as follows:
 * `[sitecore-jss]` Introduced _/editing_ submodule that contains all editing related functionality. Editing utils are now available in _/editing_ submodule. Editing utils exported from _/utils_ marked as deprecated. ([#1806](https://github.com/Sitecore/jss/pull/1806))
 * `[sitecore-jss-nextjs]` EditingRenderMiddleware `resolvePageUrl` function now accepts an object `(args: { serverUrl?: string; itemPath: string }) => string` instead of multiple parameters `(serverUrl: string, itemPath: string) => string`. `serverUrl` is now optional and omitted when Metadata Edit Mode is used.
 * `[templates/nextjs]` `[sitecore-jss-nextjs]` `[sitecore-jss]` Remove Partial rendering implementation as it will not be used by Pages in its current implementation - includes removing of EditingComponentPlaceholder component, few constants associated with it and RenderingType enum ([#1821](https://github.com/Sitecore/jss/pull/1821))
+* `[sitecore-jss-angular]``[sitecore-jss-schematics]``[templates/angular]` Upgrade Angular to v17 ([#1837](https://github.com/Sitecore/jss/pull/1837))
+  * Update Angular and core dependencies to ~17.3.11
```
Comment (illiakovalenko) [inline CHANGELOG.md:46]:
```suggestion
  * Updated Angular and core dependencies to ~17.3.11
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,10 @@ Our versioning strategy is as follows:
...
+  * Updated Typescript to ~5.2.2
+  * Update zone.js/dist/zone-node to zone.js
```
Comment (illiakovalenko) [inline CHANGELOG.md:48]: Here it means that you replaced import statements? Can you please update the description a bit?

File: docs/upgrades/unreleased.md [hunk 3]
```diff
@@ -41,6 +41,18 @@
             configText += `config.${prop} = process.env.${constantCase(prop)} || "${config[prop]?.toString().trim()}";\n`;
         ```
 
+* Update import in _src/templates/angular/server.bundle.ts_
```
Comment (illiakovalenko) [inline docs/upgrades/unreleased.md:44]: Customer will need to update Angular and Typescript as well
Can you add these steps as well?

**Review by illiakovalenko — APPROVED** [review]
Looks good! Approved with few comments

File: CHANGELOG.md (hunk 2 above)
Comment (illiakovalenko) [reply CHANGELOG.md:48]:
```suggestion
  * Updated import statements from zone.js/dist/zone-node to zone.js
```

File: docs/upgrades/unreleased.md (hunk 3 above)
Comment (illiakovalenko) [reply docs/upgrades/unreleased.md:48]: Updated

**Pull request merged**